sql/core/src/test/scala/org/apache/spark/sql/execution/command/v2/DescribeTableSuite.scala

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.command.v2
+
+import org.apache.spark.sql.{AnalysisException, Row}
+import org.apache.spark.sql.connector.catalog.TableCatalog
+import org.apache.spark.sql.execution.command
+import org.apache.spark.sql.types.{BooleanType, MetadataBuilder, StringType, StructType}
+import org.apache.spark.util.Utils
+
+/**
+ * The class contains tests for the `DESCRIBE TABLE` command to check V2 table catalogs.
+ */
+class DescribeTableSuite extends command.DescribeTableSuiteBase with CommandSuiteBase {
+  override def namespace: String = "ns1.ns2"
+
+  test("DESCRIBE TABLE with non-'partitioned-by' clause") {
+    withNamespaceAndTable(namespace, "table") { tbl =>
+      spark.sql(s"CREATE TABLE $tbl (id bigint, data string) $defaultUsing " +
+        "CLUSTERED BY (id) INTO 3 BUCKETS")
+      val descriptionDf = spark.sql(s"DESCRIBE TABLE $tbl")
+      assert(descriptionDf.schema.map(field => (field.name, field.dataType)) ===
+        Seq(
+          ("col_name", StringType),
+          ("data_type", StringType),
+          ("comment", StringType)))
+      val description = descriptionDf.collect()
+      assert(description === Seq(
+        Row("data", "string", null),
+        Row("id", "bigint", null),
+        Row("", "", ""),
+        Row("# Partitioning", "", ""),
+        Row("Part 0", "bucket(3, id)", "")))
```

Comment:
In fact, partition transforms are implicitly named. e.g. people can do `TRUNCATE TABLE t PARTITION (a=1)`. We can get the partition names if the table implements `SupportsPartitionManagement`.

Now I have a new proposal to display partitionings, which is more consistent between v1 and v2
```
# Partition Information	                    	                    
# col_name          	data_type           	comment             
c                   	string              	                    
d                   	string
# Bucketing Information
# bucket_cols         sort_cols                num_buckets
a, b                    c, d                        4
```
If there exists partition transforms
```
# Partition Information	                    	                    
# col_name          	transform
x                         year(ts)
y                         bucket(3, id)
z                         identity(d)
```
If the partition col names are unknown (the table doesn't implement `SupportsPartitionManagement`), we can use `unknown_name_0`, `unknown_name_1`, etc.